apps/server/src/shared/repo/task/task.repo.ts

```diff
@@ -64,7 +64,10 @@ export class TaskRepo {
 		scope.addQuery(allForFinishedCoursesAndLessons.query);
 		scope.addQuery(allForCreator.query);
 
-		const order = { dueDate: SortOrder.desc };
+		// The dueDate can be similar to solve pagination request missmatches we must sort it over id too.
+		// Because after executing limit() in mongoDB it is resort by similar dueDates.
+		// It exist indexes for dueDate and for _id but no combined index, because it is to expensive for only small performance boost.
+		const order = { dueDate: SortOrder.desc, _id: 1 };
```

Comment:
is **_id** correctly  or **id**?

Reply:
Good question, _id is what the DB has and the entity know id and _id. But DB calls for searching id also works and managed by the enity manager.

Reply:
```suggestion
		const order = { dueDate: SortOrder.desc, _id: SortOrder.asc };
```
?

Reply:
The ORM should be able to map the `id` property for the native query like in this MikroORM test:
https://github.com/mikro-orm/mikro-orm/blob/539311efe4d450c48fb0be2aff372bd5b64dd483/tests/EntityManager.mysql.test.ts#L1929

Reply:
updated